src/main/java/frc/robot/subsystems/pivot/Pivot.java

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems.pivot;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.utils.debugging.LoggedTunableNumber;
+import java.util.function.Supplier;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+public class Pivot extends SubsystemBase {
+  /** List of position setpoints for the pivot */
+  public enum PivotGoal {
+    // NOTE These setpoints must be updated per robot, "DEMO" is merely an example setpoint.
+
+    DEMO(() -> Rotation2d.fromDegrees(10.0));
+
+    // Suppliers are used to keep in line with this team's programming convention. In some
+    // instances, we may need to have a goal that dynamically changes based on a state from another
+    // class. If we do not use a Supplier, java will only create an instance of the setpoint once
+    // and never check if it is updated again.
+    Supplier<Rotation2d> goal;
+
+    PivotGoal(Supplier<Rotation2d> goal) {
+      this.goal = goal;
+    }
+
+    public Rotation2d getGoal() {
+      return this.goal.get();
+    }
+  }
+
+  // Hardware is where we interact with the motors and sensors directly
+  private final PivotKrakenHardware kPivotHardware;
```

Comment:
This would have an 'm' prefix, not 'k', since its state can change.  Same with kPivotInputs.